### PR TITLE
[C-Api/Single] prepare no-alloc invoke

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Code clean, prepare new API to invoke the model with allocated output data handle.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-single.c [hunk 1]
```diff
@@ -997,7 +1034,8 @@ ml_single_invoke (ml_single_h single,
   }
 
   if (single_h->ignore_output == FALSE) {
-    *output = single_h->output;
+    if (need_alloc)
```
Comment (myungjoo) [inline api/capi/src/nnstreamer-capi-single.c:1037]: At Line 1029. ```__invoke (single_h) ``` calls ```ml_tensors_data_clone_no_alloc ()```, which does ```single_h->output = NULL;```. Deleting the given output data. You may need to fix it before merging this for no-alloc.

**Comment by jaeyun-jung** [comment]
PR updated (2020-09-21)
- add flag about output allocation

File: api/capi/src/nnstreamer-capi-single.c (hunk 1 above)
Comment (jaeyun-jung) [reply api/capi/src/nnstreamer-capi-single.c:1047]: I added new flag about allocation.

**Review by myungjoo — APPROVED** [review]
LGTM! Let's apply to the product! :)

**Comment by myungjoo** [comment]
Anyway, don't you need this for 1.6.0? (performance fix for the product)

**Review by again4you — APPROVED** [review]
LGTM!

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**